src/main/java/org/broadinstitute/barclay/argparser/TaggedArgumentParser.java

```diff
@@ -0,0 +1,333 @@
+        // middle to make visual key inspection easier
+        return rawOptionString + ARGUMENT_TAG_NAME_SEPARATOR + rawArgumentValue;
+    }
+
+    /**
+     * Parse a tag string and populate a TaggedArgument with values.
+     * @param tagString tag string (including logical name and attributes, no option name)
+     * @param taggedArg TaggableArgument to receive tags
+     */
+    public void populateArgumentTags(final TaggedArgument taggedArg, final String tagString) {
+        final ParsedArgument pa = ParsedArgument.of(tagString);
+        taggedArg.setTag(pa.getName());
+        taggedArg.setTagAttributes(pa.keyValueMap());
+    }
+
+    /**
+     * Given a TaggedArgument implementation, return a string representation of the tags and attributes
+     * for display purposes. The string returned does not include the argument name.
+     *
+     * @param taggedArg implementation of TaggedArgument interface
+     * @return a display string representing the tag name and attributes
+     */
+    public static String getDisplayString(final TaggedArgument taggedArg) {
+        Utils.nonNull(taggedArg);
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(ARGUMENT_TAG_NAME_SEPARATOR);
+        sb.append(taggedArg.getTag() == null ? "NoTagName" : taggedArg.getTag());
+
+        // check for nulls since we can't assume the implementation is compliant
+        if (taggedArg.getTagAttributes() != null) {
+            taggedArg.getTagAttributes().entrySet().stream()
+                    .filter(entry -> entry.getValue() != null)
+                    .forEach(
+                            entry -> {
+                                sb.append(ARGUMENT_KEY_VALUE_PAIR_DELIMITER);
+                                sb.append(entry.getKey().toString());
+                                sb.append(ARGUMENT_KEY_VALUE_SEPARATOR);
+                                sb.append(entry.getValue().toString());
+                            });
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Represents a parsed, tagged argument.
+     *
+     * May have attributes.
+     */
+    private static final class ParsedArgument{
+        private final String name;
+        private final Map<String, String> keyValueMap;
+
+        /**
+         * Parses an argument value String of the forms:
+         *
+         * "logical_name(,key=value)"
+         *
+         * into logical name and key=value pairs.
+         *
+         * @param rawTagValue argument value from the command line to parse
+         * @return The argument parsed from the provided string.
+         */
+        public static ParsedArgument of(final String rawTagValue) {
+            final String[] tokens = rawTagValue.split(ARGUMENT_KEY_VALUE_PAIR_DELIMITER);
+            if ( Arrays.stream(tokens).anyMatch(String::isEmpty)) {
+                throw new CommandLineException.BadArgumentValue("", rawTagValue, "Empty tag name encountered. " + USAGE);
```

Comment:
It's not necessarily an empty tag name -- it's either an empty tag or an empty key/value pair.